core/commands/pubsub.go

```diff
@@ -0,0 +1,139 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	cmds "github.com/ipfs/go-ipfs/commands"
+
+	floodsub "gx/ipfs/QmQriRMW5cCJyLrzDnXi7fZ5mVbetiEZjPjbqoJhuSL94m/floodsub"
+	u "gx/ipfs/QmZNVWh8LLjAavuQ2JXuFmuYH3C11xo988vSgp7UQrTRj1/go-ipfs-util"
+)
+
+var PubsubCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Only pubsub by the vaguest technical definition",
+	},
+	Subcommands: map[string]*cmds.Command{
+		"pub": PubsubPubCmd,
+		"sub": PubsubSubCmd,
+	},
+}
+var PubsubSubCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "subscribe to messages on a given topic",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("topic", true, false, "String name of topic to subscribe to."),
```

Comment:
options for other formats? as discussed in https://github.com/libp2p/pubsub-notes/blob/master/flooding/cli-ux